File: src/main/java/silly/yet/swapping/SillySwapping.java
package silly.yet.swapping;

import net.minecraft.client.Minecraft;
import net.minecraft.client.entity.EntityPlayerSP;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.item.ItemStack;
import net.minecraftforge.fml.common.eventhandler.EventPriority;
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
import net.minecraftforge.fml.common.gameevent.TickEvent;
import silly.yet.swapping.config.SillyConfig;

public class SillySwapping {

    private int originalSlot = -1; // Store the original slot
    private int targetSlot = -1; // Store the slot with Etherwarp Conduit
    private boolean keyHeld = false;
    private boolean active = false;


    @SubscribeEvent(priority = EventPriority.NORMAL, receiveCanceled = true)
    public void onTick(TickEvent.PlayerTickEvent event) {
        if (!SillyConfig.sillySwapper) return;
        EntityPlayerSP player = Minecraft.getMinecraft().thePlayer;
        if (player == null) return;
        ItemStack heldItem = player.inventory.getCurrentItem();
        if (!player.worldObj.isRemote) return;

        boolean keyHeld = SillyConfig.sillyBind.isActive();
        if (keyHeld == this.keyHeld) return;
        this.keyHeld = keyHeld;

        int currentSlot = player.inventory.currentItem;
        if (keyHeld) {
            if (SillyConfig.AOTVMode && heldItem != null && heldItem.getDisplayName().contains("Aspect of the Void")) {
                if (Minecraft.getMinecraft().currentScreen != null) return;
                targetSlot = findHotbarSlotWithItem(player, "Etherwarp Conduit");
                if (targetSlot == -1) return;
                originalSlot = currentSlot;
                player.inventory.currentItem = targetSlot;
                active = true;
                Minecraft.getMinecraft().thePlayer.inventory.currentItem = targetSlot;
            } else if (!SillyConfig.AOTVMode) {
                if (Minecraft.getMinecraft().currentScreen != null) return;
                targetSlot = findHotbarSlotWithItem(player, "Etherwarp Conduit");
                if (targetSlot == -1) return;
                originalSlot = currentSlot;
                player.inventory.currentItem = targetSlot;
                active = true;
                Minecraft.getMinecraft().thePlayer.inventory.currentItem = targetSlot;
            }
        } else {
            boolean toggle = active;
            active = false;
            if (heldItem != null && toggle && heldItem.getDisplayName().contains("Etherwarp Conduit")) {
                player.inventory.currentItem = originalSlot;
                Minecraft.getMinecraft().thePlayer.inventory.currentItem = originalSlot;
            }
        }
    }

    private int findHotbarSlotWithItem(EntityPlayer player, String itemName) {
        for (int i = 0; i < 9; i++) {
            ItemStack stack = player.inventory.getStackInSlot(i);
            if (stack != null && stack.hasDisplayName() && stack.getDisplayName().contains(itemName)) {
                return i;
            }
        }
        return -1;
    }
}

File: src/main/java/silly/yet/swapping/SillySwapping.kt
package silly.yet.swapping

import net.minecraft.client.Minecraft
import net.minecraft.entity.player.EntityPlayer
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
import net.minecraftforge.fml.common.gameevent.TickEvent.PlayerTickEvent
import silly.yet.swapping.config.SillyConfig

class SillySwapping {

    private var originalSlot = 1
    private var keyHeld = false
    private var active = false

    private val mc get() = Minecraft.getMinecraft()

    @SubscribeEvent
    fun onTick(event: PlayerTickEvent) {
        if (!SillyConfig.sillySwapper) return

        val player = mc.thePlayer ?: return

        val heldItem = player.inventory.getCurrentItem()

        val isActive = SillyConfig.sillyBind.isActive
        if (isActive == keyHeld) return
        keyHeld = isActive

        val currentSlot = player.inventory.currentItem
        if (isActive) {
            if (!SillyConfig.AOTVMode || heldItem != null && heldItem.displayName.contains("Aspect of the Void")) {
                val targetSlot = player.findHotbarSlotWithItem("Etherwarp Conduit") ?: return

                originalSlot = currentSlot
                player.setSlot(targetSlot)
                active = true
            }
        } else {
            if (heldItem != null && active && heldItem.displayName.contains("Etherwarp Conduit")) {
                player.setSlot(originalSlot)
            }
            active = false
        }
    }

    private fun EntityPlayer.findHotbarSlotWithItem(itemName: String): Int? {
        for (i in 0..8) {
            val stack = inventory.getStackInSlot(i) ?: continue
            if (stack.hasDisplayName() && stack.displayName.contains(itemName)) {
                return i
            }
        }
        return null
    }

    private fun EntityPlayer.setSlot(slot: Int) {
        if (slot in 0..8) inventory.currentItem = slot
    }
    
}